apis/components/v1/kserve_types.go

```diff
@@ -17,20 +17,49 @@ limitations under the License.
 package v1
 
 import (
-	"github.com/opendatahub-io/opendatahub-operator/v2/apis/components"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/components"
+	infrav1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/infrastructure/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+const (
+	KserveComponentName = "kserve"
+	// value should match what's set in the XValidation below
+	KserveInstanceName = "default-kserve"
+	KserveKind         = "Kserve"
+)
+
+// +kubebuilder:validation:Pattern=`^(Serverless|RawDeployment)$`
+type DefaultDeploymentMode string
+
+const (
+	// Serverless will be used as the default deployment mode for Kserve. This requires Serverless and ServiceMesh operators configured as dependencies.
+	Serverless DefaultDeploymentMode = "Serverless"
+	// RawDeployment will be used as the default deployment mode for Kserve.
```

Comment:
I'm thinking about this comment :thinking: I just copied it from where it was before. Are you suggesting to remove it because RawDeployment is not the "default"? I think the comment might still be valid for this constant though, since the type is "DefaultDeploymentMode", so if this const is being used, then it's the "default deployment mode".

Reply:
on incubation this most likely is a copy-paste error, i will remove it from there first.
you can keep it here, we can rebase from incubatoin later

Reply:
Sounds good, thanks!